cdap-app-templates/cdap-etl/cdap-etl-api-spark/src/main/java/co/cask/cdap/etl/api/batch/SparkMLLibContext.java

```diff
@@ -47,11 +48,12 @@
    * @param datasetName the name of the {@link Dataset} to be read as an RDD
    * @param kClass      the key class
    * @param vClass      the value class
-   * @param <T>         type of RDD
+   * @param <K>        type of RDD key
+   * @param <V>        type of RDD value
    * @return the RDD created from Dataset
    * @throws UnsupportedOperationException if the SparkContext is not yet initialized
    */
-  <T> T readFromDataset(String datasetName, Class<?> kClass, Class<?> vClass);
+  <K, V> JavaPairRDD<K, V> readFromDataset(String datasetName, Class<?> kClass, Class<?> vClass);
```

Comment:
also updated the stream `readFromStream` methods to use `LongWritable` as the key type.